src/BankTransaction.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+namespace PhpTwinfield;
+
+use PhpTwinfield\Enums\DebitCredit;
+use PhpTwinfield\Enums\Destiny;
+use PhpTwinfield\Transactions\BankTransactionLine\Base;
+use PhpTwinfield\Transactions\TransactionFields\AutoBalanceVatField;
+use PhpTwinfield\Transactions\TransactionFields\DestinyField;
+use PhpTwinfield\Transactions\TransactionFields\FreeTextFields;
+use PhpTwinfield\Transactions\TransactionFields\OfficeField;
+use PhpTwinfield\Transactions\TransactionFields\StartAndCloseValueFields;
+use PhpTwinfield\Transactions\TransactionLineFields\PeriodField;
+use Webmozart\Assert\Assert;
+
+class BankTransaction
+{
+    use DestinyField;
+    use AutoBalanceVatField;
+    use PeriodField;
+    use OfficeField;
+
+    use StartAndCloseValueFields;
+
+    use FreeTextFields;
+
+    /**
+     * The date/time on which the transaction was created.
+     * Read-only attribute.
+     *
+     * @var \DateTimeImmutable
+     */
+    private $inputDate;
+
+    /**
+     * Transaction type code.
+     *
+     * @var mixed
+     */
+    private $code;
+
+    /**
+     * Transaction number.
+     * When creating a new bank transaction, don't include this tag as the transaction number is determined by the system. When updating a bank transaction, the related transaction number should be provided.
+     *
+     * @var int
+     */
+    private $number;
+
+    /**
+     * Transaction date.
+     * Optionally, it is possible to suppress warnings about 'date out of range for the given period' by adding the raisewarning attribute and set its value to false. This overwrites the value of the raisewarning attribute as set on the root element.
+     *
+     * @var \DateTimeInterface
+     */
+    private $date;
+
+    /**
+     * Number of the bank statement. Don't confuse this number with the transaction number.
+     *
+     * @var int
+     */
+    private $statementnumber;
+
+    /**
+     * The bank transaction origin.
+     * Read-only attribute.
+     *
+     * @var mixed
+     */
+    private $origin;
+
+    /**
+     * The date/time on which the bank transaction was modified the last time.
+     * Read-only attribute.
+     *
+     * @var \DateTimeInterface
+     */
+    private $modificationDate;
+
+    /**
+     * @var Transactions\BankTransactionLine\Base[]
+     */
+    private $transactions = [];
+
+    /**
+     * The bank transaction origin.
+     * Read-only attribute.
+     *
+     * @return mixed
+     */
+    public function getOrigin()
```

Comment:
Without a setter function for read-only attributes there's also no way for a mapper to set the attributes. But there's currently no mapper to get these kind of transactions from Twinfield, right? So the setters aren't really needed yet.

Reply:
Yes, my thoughts exactly.